components/stats/ExplorerDropdown.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { ExternalLink, ChevronDown } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { BlockExplorer } from "@/types/stats";
+
+interface ExplorerDropdownProps {
+  explorers?: BlockExplorer[];
+  size?: "sm" | "default";
+  variant?: "outline" | "default" | "ghost";
+  showIcon?: boolean;
+  buttonText?: string;
+}
+
+export function ExplorerDropdown({
+  explorers,
+  size = "sm",
+  variant = "outline",
+  showIcon = true,
+  buttonText = "View Explorer",
+}: ExplorerDropdownProps) {
+  // No explorers available
+  if (!explorers || explorers.length === 0) {
+    return null;
+  }
+
+  // Single explorer - show direct link button
+  if (explorers.length === 1) {
+    return (
+      <Button
+        variant={variant}
+        size={size}
+        onClick={(e) => {
+          e.stopPropagation();
+          window.open(explorers[0].link, "_blank");
+        }}
+        className="flex items-center gap-1.5 text-xs whitespace-nowrap border-neutral-200 dark:border-neutral-700 text-neutral-700 dark:text-neutral-300 hover:bg-neutral-100 dark:hover:bg-neutral-800 hover:text-neutral-900 dark:hover:text-neutral-100"
+      >
+        {showIcon && <ExternalLink className="h-3 w-3" />}
+        {buttonText}
+      </Button>
+    );
+  }
+
+  // Multiple explorers - show dropdown menu
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button
+          variant={variant}
+          size={size}
+          onClick={(e) => e.stopPropagation()}
+          className="flex items-center gap-1.5 text-xs whitespace-nowrap border-neutral-200 dark:border-neutral-700 text-neutral-700 dark:text-neutral-300 hover:bg-neutral-100 dark:hover:bg-neutral-800 hover:text-neutral-900 dark:hover:text-neutral-100"
+        >
+          {showIcon && <ExternalLink className="h-3 w-3" />}
+          {buttonText}
+          <ChevronDown className="h-3 w-3" />
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end" className="w-40">
+        {explorers.map((explorer, index) => (
+          <DropdownMenuItem
+            key={index}
+            onClick={(e) => {
+              e.stopPropagation();
+              window.open(explorer.link, "_blank");
+            }}
+            className="cursor-pointer text-xs"
+          >
+            <ExternalLink className="h-3 w-3 mr-2" />
```

Comment:
pls replace this with "ArrowUpRight", same as what we use in "Submit Your L1" button.